src/server/verification/verification.js

```diff
@@ -19,49 +21,41 @@ class Verifications implements VerificationAPI {
     this.log.debug('Verifying user:', { user })
-    const sessionId = verificationData.sessionId
-    this.log.debug('sessionId:', { sessionId })
-    const searchData = Helper.prepareSearchData(verificationData)
-    // log.info('searchData', { searchData })
-    const isDuplicate = await Helper.isDuplicatesExist(searchData, verificationData.enrollmentIdentifier)
-    GunDBPublic.gun
-      .get(sessionId)
-      .get('isNotDuplicate')
-      .put(!isDuplicate) // publish to subscribers
 
-    this.log.debug('isDuplicate result:', { user: user.identifier, isDuplicate })
-    if (isDuplicate) return { ok: 1, isDuplicate }
-    const enrollData = Helper.prepareEnrollmentData(verificationData)
-    // log.info('enrollData', { enrollData })
-    const enrollResult: EnrollResult = await Helper.enroll(enrollData)
-    GunDBPublic.gun
-      .get(sessionId)
-      .get('isEnrolled')
-      .put(enrollResult.alreadyEnrolled || (enrollResult.enrollmentIdentifier ? true : false)) // publish to subscribers
-    const livenessFailed = (enrollResult && enrollResult.ok === false) || enrollResult.livenessResult === 'undetermined'
-    this.log.debug('liveness result:', { user: user.identifier, livenessFailed })
+    const enrollHandler = async (userId, sessionId, data) => {
+      this.log.debug('Verifying user enrollHandler :', data)
+      const sessionRef = GunDBPublic.session(sessionId)
+      const enrollPayload = pick(data, 'isDuplicate', 'isLive', 'isEnrolled')
 
-    GunDBPublic.gun
-      .get(sessionId)
-      .get('isLive')
-      .put(!livenessFailed) // publish to subscribers
-    if (livenessFailed) return { ok: 1, livenessPassed: false }
+      sessionRef.put(enrollPayload)
 
-    //this.log.debug('liveness result:', { user: user.identifier, livenessPassed }) // This is left to support future granularity for user better UX experience. Consider using authenticationFacemapIsLowQuality property https://dev.zoomlogin.com/zoomsdk/#/webservice-guide
-    //if (!livenessPassed) return { ok: 1, livenessPassed }
-
-    const isVerified =
-      !isDuplicate && (enrollResult.alreadyEnrolled || (enrollResult.enrollmentIdentifier ? true : false)) // enrollResult.enrollmentIdentifier should return true if there is value in it (and not the value itself) into isVerified.
-    return {
-      ok: 1,
-      isVerified,
-      enrollResult: { ...enrollResult, enrollmentIdentifier: verificationData.enrollmentIdentifier }
+      if (data.ok && data.result) {
+        this.log.debug('Whitelisting new user', user)
+        try {
+          await Promise.all([
+            AdminWallet.whitelistUser(user.gdAddress, user.profilePublickey),
+            storage
+              .updateUser({ identifier: user.loggedInAs, isVerified: true })
+              .then(updatedUser => this.log.debug('updatedUser:', updatedUser))
+          ])
+          await sessionRef.get({ isWhitelisted: true })
```

Comment:
sessionRef.put?
no need for await

Reply:
done

Reply:
@sirpy Understood what you've mean. We needed to check also humanAPI.ts source from FaceVerification. Here we can see that result is just an object containing informartion about errors & liveness leve. And you then analyze data from it and set isEnroll flag. So presense of result isn't 100% guarantee of successfull check, we should check isEnroll

@StanislavShevchenko pls look at this comment